src/pages/home/sidebar/ChatSwitcherView.js

```diff
@@ -461,12 +466,39 @@ class ChatSwitcherView extends React.Component {
             }
         }
 
-        this.setState({
-            options: Array.from(matches),
-        });
+        const options = Array.from(matches);
+        if (options.length === 0) {
+            if (value && (Str.isValidEmail(value) || Str.isValidPhone(value))) {
+                let login = value;
+                if (Str.isValidPhone(value)) {
```

Comment:
NAB this could me made into a variable `isPhone`

Reply:
I refrained from making this a variable since i don't want to call `Str.isValidPhone` unless options.length === 0 and value is not empty. So if i did,
```
isPhone = options.length === 0 && value && Str.isValidPhone(value);
```
it feels a little weird so I'm just calling `Str.isValidPhone` for now. I didn't want to do just,
```
isPhone = Str.isValidPhone(value);
```
before the if condition because i think keytyping it pretty slow and it feels like string checks on keytype would slow it down further hence i wanted to string check only when options.length === 0 and value is not empty. Leaving it as is for now but if you feel differently let me know.